File: routes/cpp-to-javascript/corpus/module/equivalence_module.cpp
#include <cstdint>

std::int64_t calculate(std::int64_t subtotal, std::int64_t tax) {
    if (subtotal < 0) {
        return 0;
    }
    return subtotal + tax;
}

std::int64_t clamp(std::int64_t value, std::int64_t minimum, std::int64_t maximum) {
    if (value < minimum) {
        return minimum;
    }
    if (value > maximum) {
        return maximum;
    }
    return value;
}

std::int64_t difference(std::int64_t left, std::int64_t right) {
    return left - right;
}

double clampNumber(double value, double minimum, double maximum) {
    if (value < minimum) {
        return minimum;
    }
    if (value > maximum) {
        return maximum;
    }
    return value;
}

bool both(bool left, bool right) {
    return left && right;
}

File: routes/javascript-to-cpp/corpus/module/equivalence_module.mjs
/**
 * @param {integer} subtotal
 * @param {integer} tax
 * @returns {integer}
 */
export function calculate(subtotal, tax) {
  if (subtotal < 0) { return 0; }
  return subtotal + tax;
}

/**
 * @param {integer} value
 * @param {integer} minimum
 * @param {integer} maximum
 * @returns {integer}
 */
export function clamp(value, minimum, maximum) {
  if (value < minimum) { return minimum; }
  if (value > maximum) { return maximum; }
  return value;
}

/**
 * @param {integer} left
 * @param {integer} right
 * @returns {integer}
 */
export function difference(left, right) {
  return left - right;
}

/**
 * @param {number} value
 * @param {number} minimum
 * @param {number} maximum
 * @returns {number}
 */
export function clampNumber(value, minimum, maximum) {
  if (value < minimum) { return minimum; }
  if (value > maximum) { return maximum; }
  return value;
}

/**
 * @param {boolean} left
 * @param {boolean} right
 * @returns {boolean}
 */
export function both(left, right) {
  return left && right;
}
